### PR TITLE
Add error handling interface

### PR DESCRIPTION
**Pull request by smartguest**
Companion PR for https://github.com/microsoft/azuredatastudio/pull/21627

### CODE REVIEW

File: src/azdata.proposed.d.ts [hunk 1]
```diff
@@ -413,6 +413,24 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+	/**
+	 * An interface for receiving diagnostic data from a provider.
```
Comment (cheenamalhotra) [inline src/azdata.proposed.d.ts:417]:
```suggestion
	 * Provides APIs for provider to run error diagnostics.
```

File: src/azdata.proposed.d.ts [hunk 2]
```diff
@@ -413,6 +413,24 @@ declare module 'azdata' {
...
+	 */
+	export interface Diagnostics {
+		handleErrorCode(errorCode: number, errorMessage: string): Thenable<diagnostics.ErrorDiagnosticsResponse>;
+	}
+
+	export namespace diagnostics {
+		export interface ErrorDiagnosticsResponse {
```
Comment (cheenamalhotra) [inline src/azdata.proposed.d.ts:424]: Please add doc comment: e.g. "Error diagnostics response containing action to be taken for received error."

File: src/azdata.proposed.d.ts [hunk 3]
```diff
@@ -413,6 +413,24 @@ declare module 'azdata' {
...
+			errorAction: string;
+		}
+
+		/**
+		 * Registers a Diagnostics object that can support a provider.
```
Comment (cheenamalhotra) [inline src/azdata.proposed.d.ts:429]:
```suggestion
		 * Registers provider with instance of Diagnostics implementation.
```

File: src/azdata.proposed.d.ts [hunk 4]
```diff
@@ -413,6 +413,24 @@ declare module 'azdata' {
...
+	/**
+	 * An interface for receiving diagnostic data from a provider.
+	 */
+	export interface Diagnostics {
+		handleErrorCode(errorCode: number, errorMessage: string): Thenable<diagnostics.ErrorDiagnosticsResponse>;
+	}
+
+	export namespace diagnostics {
+		export interface ErrorDiagnosticsResponse {
+			errorAction: string;
```
Comment (cheenamalhotra) [inline src/azdata.proposed.d.ts:425]: Can this be an enum instead, with 0 pointing to default action, i.e. to show error dialog.

**Review by kburtram — CHANGES_REQUESTED** [review]
This PR appears to only be updating a TS typings file, which is a build-time only artifact.  If these types aren't referenced by a .TS file then there should be no impact on the complied output.  So I'm not clear on the intent of this change.

**Pull request closed without merging**